hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/PartialUpdateReaderContext.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table;
+
+import org.apache.hudi.common.engine.HoodieReaderContext;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.storage.StoragePath;
+
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.Serializable;
+
+public class PartialUpdateReaderContext implements Serializable {
+
+  // Prepare broadcast variables.
+  public void prepareReaderContext() {
+    // NO operation.
+  }
+
+  // Create reader context based on broadcast variables.
+  public Option<HoodieReaderContext> getReaderContext(StoragePath basePath) {
+    return Option.empty();
+  }
+
+  // Fetch broadcast storage config.
+  public Option<Configuration> getStorageConfig() {
+    return Option.empty();
+  }
```

Comment:
I think it's better to put these new APIs to the `EngineContext` class.

Reply:
If we could not easily make such changes, I think it's OK to leave it this way now for supporting compaction with partial updates.

Reply:
`EngineContext` is not a suitable place to put these things too, maybe we just pass around the `ReaderContext` to the `#compact` method directly.